SonarQube.Common/ProjectInfo/ProjectInfoExtensions.cs

```diff
@@ -70,6 +70,26 @@ public static bool TryGetAnalysisSetting(this ProjectInfo projectInfo, string id
         }
 
         /// <summary>
+        /// Attempts to find and return the global analysis setting with the specified id
+        /// </summary>
+        /// <returns>True if the setting was found, otherwise false</returns>
+        public static bool TryGetGlobalAnalysisSetting(this ProjectInfo projectInfo, string id, out AnalysisSetting result)
+        {
+            if (projectInfo == null)
+            {
+                throw new ArgumentNullException("projectInfo");
+            }
+
+            result = null;
+
+            if (projectInfo.GlobalAnalysisSettings != null)
+            {
+                result = projectInfo.GlobalAnalysisSettings.FirstOrDefault(ar => id.Equals(ar.Id, StringComparison.Ordinal));
```

Comment:
This comparison should use the new constant (as should the equivalent TryGetAnalysisSetting method). I'll change it.